### ISSUE TITLE
Add Advanced Search options for Land Managers

### ISSUE THREAD

**Issue by kkemp85** [legiongis/hmsf-md#124]
Now that the export button exists, I started to try out advanced search options in a land manager account just to see how we would inform them to export scout reports for their sites once that is up and running. From what I can tell, this is not even possible the way the search is set up now and will require us to create some additional search options. This is an issue for two main reasons. 1. All accounts can see all scout reports no matter who submitted them (I do not think we should change this though) and 2. The only thing that connects archaeological sites with corresponding scout reports is the Site ID.

For example, If Tomoka State Park wanted to see all reports submitted on the 11 arch sites within their park, they can't just filter by "Scout Report" because they will see all scout reports submitted for all sites in Arches. If they added the Advanced Search for "Visiting Scout ID" and put in their account name they would be able to pull any reports they have submitted, but would not be able to see reports that other scouts or FPAN staff have submitted (so they wouldn't be seeing a full report of monitoring at their sites). This will be a big issue for them, especially since FPAN has been tasked to help them with their management responsibilities. If they knew the scout IDs of other people who had monitored within their park they could add them to the filter, but the chance of that is very slim. The only other option to get the search to show up with all scout reports for their sites is to search each site within their park individually to pull up reports, but this is not a good option and would take forever, even with only 11 sites.

So from what I can tell, we need to add an extra search filter that will allow Land Managers to filter Scout Reports with the "Located In" information that exists for each archaeological site. That way, Tomoka State Park could just go in and search on the Scout Report Advanced Search tab for "Managed Area Name" and put in Tomoka State Park and all reports associated with sites tagged as being in Tomoka State Park would come up. This will also be an issue with the regional level accounts. Since scout reports are for now only tied to the arch site data by site ID, this would require us to somehow auto-fill or tag the "Located in" information to a Scout Report when it is created to correlate with that site.

So, in summary, I think we need to be able to use the locational data tied to arch sites to search for Scout Reports if that makes sense. Is this possible? Or do you have a better suggestion? We need to make it simple with just a few clicks for land managers to filter through all 1400 scout reports and be able to export all scout reports for only sites on their lands.

**Comment by mradamcox** [comment]
@kkemp85 Completely understand this. May try to address it with a new button, will think about it.

**Comment by kkemp85** [comment]
@mradamcox This issue looks like it is going to be bigger than just a problem for land managers. One of our FPAN staff emailed me today to ask how she can search Arches to see how many shipwreck sites have Scout Reports filled out for them. I realized that there isn't a way to do this. There isn't even a way for me to get a list of Scout Reports per resource (structures, or cemeteries, or arch sites), much less a specific site type within a resource. This is something that will be important for normal operations but will be essential when we get to our reporting phases of the grant. I guess I hadn't really tried to do a search like this since we separated the Scout Report into its own resource.

It's so important that Scout Reports are separate resources but we have to figure out a way to tie these back to the specific site resource information so we can search through them using site data. A lot of the searches we do within the arch site, cemetery, and structure resources are to eventually get to what sites have scout reports filled out for them. 

Not sure what the solution is here or whether this comment should be a separate issue since it isn't necessarily specific to land managers. Would mean we wouldn't just need to tie locational data to scout reports as I thought before, but also site-specific info (detailed description info - site type, culture, human remains) and the FMSF name (right now it's set where you can only search by specific site name, you can't even type in "wreck" and pull up all scout reports with the word "wreck" in the site name).

**Comment by mradamcox** [comment]
After a bit of discussion, sounds like a good direction to go with this is the addition of a button in the search interface that says something like "View Reports for Current Result." It would swap the current results with only reports that are attached to the current results.
